src/com/esotericsoftware/reflectasm/FieldAccess.java

```diff
@@ -572,5 +574,25 @@ static private MethodVisitor insertThrowExceptionForFieldType (MethodVisitor mv,
 		mv.visitInsn(ATHROW);
 		return mv;
 	}
-
+	
+	static public  FieldAccess getAccessUnsafe(Class<?> type)

```

Comment:
I'd prefer `new FieldAccessUnsafe(Class)` over putting unsafe related code in FieldAccess.

Reply:
Unsafe would need to be completely separate and accessed in a way that still works when unsafe is not available. However, with unsafe being unsupported in future Java versions, I'm not sure it's worth it.